CryptoScan.Web/Main/Common/Http.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Text.Json;
+
+namespace CryptoScan.Web.Main.Common;
+
+internal static class Http
+{
+  public static async Task<ActionResult<T>> Get<T>(string url, string? notFoundError = null, string? badRequestError = null)
+  {
+    try
+    {
+      string response = await new HttpClient()
+      .GetStringAsync(url);
+
+      var options = new JsonSerializerOptions
+      {
+        PropertyNameCaseInsensitive = true
+      };
+
+      var content = JsonSerializer.Deserialize<T>(response, options);
+
+      return content == null
+        ? new NotFoundObjectResult(notFoundError)
+        : content;
+    }
+    catch (Exception ex)
```

Comment:
I'm not sure if converting every exception to badrequest is a good idea, might complicate debugging

Reply:
If you want, I can mark this as a bug, but I think it has a low priority for now.